vacasa/connect/connect.py

```diff
@@ -732,6 +732,73 @@ def create_reservation(self,
 
         return self._post(url, json={'data': {'attributes': payload}}, headers=headers).json()
 
+    def cancel_reservation(self,
+                           unit_id: int,
+                           arrival: str,
+                           departure: str,
+                           adults: int,
+                           children: int,
+                           pets: int,
+                           trip_protection: bool,
+                           discount_id: str,
+                           first_name: str,
+                           last_name: str,
+                           phone: str,
+                           rent: int,
+                           fee_amount: int,
+                           cleaning_fees: int,
+                           trip_protection_fee: int,
+                           total: int,
+                           tax_amount: int,
+                           ) -> dict:
+        """
+        :param unit_id: A Vacasa Unit ID
+        :param arrival: Checkin date in 'YYYY-MM-DD' format
+        :param departure: Checkout date in 'YYYY-MM-DD' format
+        :param adults: How many adults will be staying
+        :param children: How many children will be staying
+        :param pets: How many pets will be staying
+        :param trip_protection: Has the user requested trip protection?
+                -1 No
+                0 TBD
+                1 Yes
+        :param discount_id:
+        :param first_name:
+        :param last_name:
+        :param phone:
+        :param rent:
+        :param fee_amount:
+        :param cleaning_fees:
+        :param trip_protection_fee:
+        :param total:
+        :param tax_amount:
+        :return:
```

Comment:
*pedantry alert*
I prefer the google style guide docstring format (https://google.github.io/styleguide/pyguide.html#383-functions-and-methods - see the example right below this section), the default format is a little punctuation/line noise heavy.  I think there's an intellij setting to switch formats & make it easy for you.